HOMEWORK v16

Zbyslav opened the pull request:
Methods for random letters_ and numbers_blocks added

Zbyslav committed:
Create homework.rb

Zbyslav committed:
Create homework.rb

sveredyuk commented on homework.rb:
```diff
@@ -1,6 +1,20 @@
+ALPHABET = ("A".."Z").to_a
+
+def rand_letters_block
+	letters_block = Array.new
```
2 spaces

sveredyuk commented on homework.rb:
```diff
@@ -1,6 +1,20 @@
+	letters_block = Array.new
+	4.times do letters_block << ALPHABET[rand(ALPHABET.length)]
+	end
+	return letters_block.join
```
In Ruby method returns always the last line, so `return` here is redundant

Zbyslav committed:
Create homework.rb

sveredyuk commented on homework.rb:
```diff
@@ -1,6 +1,20 @@
+
+def rand_letters_block
+  letters_block = Array.new
+  4.times do letters_block << ALPHABET[rand(ALPHABET.length)]
```
4 - magic number.. 5 also

sveredyuk replied:
One line style without `end`
4.times do { letters_block << ALPHABET[rand(ALPHABET.length)] }

sveredyuk commented on homework.rb:
```diff
@@ -1,6 +1,20 @@
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
+  rand_letters_block + '-' + rand_letters_block + '-' + rand_numbers_block + '-' + rand_numbers_block
```
use interpolation instead plus

Zbyslav committed:
Create homework.rb

- magic numbers removed
- `.times` rewrote in one line
- used interpolation in `generate`
